lualib/resty/kong/var.lua

```diff
@@ -0,0 +1,161 @@
+local ffi = require "ffi"
+local base = require "resty.core.base"
+
+
+local C = ffi.C
+local ffi_new = ffi.new
+local ffi_str = ffi.string
+local type = type
+local error = error
+local tostring = tostring
+local getmetatable = getmetatable
+local get_request = base.get_request
+local get_size_ptr = base.get_size_ptr
+local subsystem = ngx.config.subsystem
+
+
+local variable_index = {}
+
+local ngx_lua_kong_ffi_var_get_by_index
+local ngx_lua_kong_ffi_var_set_by_index
+local ngx_lua_kong_ffi_var_load_indexes
+
+
+if subsystem == "http" then
+    ffi.cdef[[
+    int ngx_http_lua_kong_ffi_var_get_by_index(ngx_http_request_t *r,
+        unsigned int var_index, char **value, size_t *value_len, char **err);
+
+    int ngx_http_lua_kong_ffi_var_set_by_index(ngx_http_request_t *r,
+        unsigned int index, const unsigned char *value, size_t value_len,
+        char **err);
+
+    int ngx_http_lua_kong_ffi_var_load_indexes(ngx_str_t **names,
+        unsigned int *count, char **err);
+    ]]
+
+    ngx_lua_kong_ffi_var_get_by_index = C.ngx_http_lua_kong_ffi_var_get_by_index
+    ngx_lua_kong_ffi_var_set_by_index = C.ngx_http_lua_kong_ffi_var_set_by_index
+    ngx_lua_kong_ffi_var_load_indexes = C.ngx_http_lua_kong_ffi_var_load_indexes
+end
+
+
+local value_ptr = ffi_new("unsigned char *[1]")
+local errmsg = base.get_errmsg_ptr()
+
+local function load_indexes(count)
+    count = count or 100
+    local names_buf = ffi_new("ngx_str_t *[?]", count)
+    local count_ptr = ffi_new("unsigned int [1]")
+    count_ptr[0] = count
+
+    local rc = ngx_lua_kong_ffi_var_load_indexes(names_buf,
+                                                 count_ptr, errmsg)
+
+    if rc == 0 then -- NGX_OK
```

Comment:
We should use `ngx.OK`, `ngx.ERR` (after module caching), etc for those core constants. Same below.